odoo/addons/base/models/ir_ui_view.py

```diff
@@ -431,13 +431,34 @@ def _valid_inheritance(self, arch):
                         self._raise_view_error(message, node)
         return True
 
+    def _validate_render_report(self):
+        for view in self:
+            if view.type != 'qweb':
+                return
+            arch_value = 't-call="%s"' % view.xml_id
+            parent_views = self.env['ir.ui.view'].search([('type', '=', 'qweb'), ('arch_db', 'ilike', arch_value)])
+            if parent_views:
+                parent_views._validate_render_report()
```

Comment:
i would improve the recursivity, or remove it
either return the recursive statement
```
if parent views:
    return parent_views._validate_render_report()
```
(since it's unlikely that any view would be used in more than one report, and if it is you probably only want to test it once anyway)
or do sdomething like this to accumulate all parent views in a single recordset and then try to find a report that calls one of these views and render it
```py
to_validate = self
parent_views = self.env['ir.ui.view'].search([('type', '=', 'qweb'), ('arch_db', 'ilike', arch_value)])
while parent_views:
     to_validate  = to_validate | parent_views
     t_call_domain = expression.OR([('arch_db', 'ilike', f't-call="{view.xml_id}"') for view in to_validate])
     not_in_domain = [('id', 'not in', to_validate.ids)]
     domain = not_in_domain + t_call_domain +([('type', '=', 'qweb')]
    parent_views = self.env['ir.ui.view'].search([('type', '=', 'qweb'), ('arch_db', 'ilike', arch_value)])
```

The first one is probably easier to read if one is used to recursion.